subscription-service/src/main/kotlin/com/egm/stellio/subscription/service/NotificationService.kt

```diff
@@ -34,7 +34,13 @@ class NotificationService(
         either {
             val id = ngsiLdEntity.id
             val types = ngsiLdEntity.types
-            subscriptionService.getMatchingSubscriptions(id, types, updatedAttributes)
+            subscriptionService.getMatchingSubscriptions(id, updatedAttributes)
+                .filter {
+                    subscriptionService.isMatchingTypesQuery(
```

Comment:
no other solution than doing an extra query for types? :(

Reply:
otherwise I don't see how to retrieve the type of the entities part of a subscription to use the function `buildTypeQuery`